### PR TITLE
Return success status in low-level triangulation functions

### PR DESCRIPTION
**Pull request by ahojnnes**
More robustly handle degenerate configurations. Noticed a rare edge case during some tests. Should barely make a difference in practice but avoids issues when using the low-level functions in new places.

### CODE REVIEW

File: src/colmap/geometry/triangulation.cc [hunk 1]
```diff
@@ -37,63 +37,58 @@
...
-    const std::vector<Eigen::Vector2d>& points2) {
-  THROW_CHECK_EQ(points1.size(), points2.size());
-
-  std::vector<Eigen::Vector3d> points3D(points1.size());
-
-  for (size_t i = 0; i < points3D.size(); ++i) {
-    points3D[i] = TriangulatePoint(
-        cam1_from_world, cam2_from_world, points1[i], points2[i]);
+  const Eigen::JacobiSVD<Eigen::Matrix4d> svd(A, Eigen::ComputeFullV);
+  if (svd.info() != Eigen::Success || svd.matrixV()(3, 3) == 0) {
```
Comment (sarlinpe) [inline src/colmap/geometry/triangulation.cc:54]: info() is only available in Eigen >= 3.4.0: https://gitlab.com/libeigen/eigen/-/commit/5bbc9cea93ef29cee2b8ffb2084d4ebca32600ba

Comment (ahojnnes) [reply src/colmap/geometry/triangulation.cc:54]: Thanks, didn't know. Will fix.

**Review by sarlinpe — APPROVED** [review]
